app/src/main/java/org/schabi/newpipe/MainActivity.java

```diff
@@ -795,9 +802,24 @@ private void handleIntent(final Intent intent) {
             ErrorActivity.reportUiError(this, e);
         }
     }
-    /*
-     * Utils
-     * */
+
+    private void setupBroadcastReceiver() {
+        broadcastReceiver = new BroadcastReceiver() {
+            @Override
+            public void onReceive(final Context context, final Intent intent) {
+                if (intent.getAction().equals(VideoDetailFragment.ACTION_PLAYER_STARTED)) {
+                    final Fragment fragmentPlayer = getSupportFragmentManager()
+                            .findFragmentById(R.id.fragment_player_holder);
+                    if (fragmentPlayer == null) {
+                        NavigationHelper.showMiniPlayer(getSupportFragmentManager());
+                    }
```

Comment:
Could you add a comment explaining what this piece of code does? 
Is the following interpretation correct? "If a player is started when there is no `VideoDetailFragment` active, the bottom sheet is initialized."

Reply:
Added comments with some contextual explanation